pulp_rpm/app/models/package.py

```diff
@@ -231,30 +253,34 @@ class Package(Content):
     is_modular = models.BooleanField(default=False)
 
     # createrepo_c treats 'nosrc' arch (opensuse specific use) as 'src' so it can seem that two
-    # packages are the same when they are not. By adding 'location_href' here we can recognize this.
+    # packages are the same when they are not. By adding 'filename' here we can recognize this.
     # E.g. glibc-2.26.11.3.2.nosrc.rpm vs glibc-2.26.11.3.2.src.rpm
-    repo_key_fields = ("name", "epoch", "version", "release", "arch", "location_href")
-
-    @property
-    def filename(self):
-        """
-        Create a filename for an RPM based upon its NEVRA information.
-        """
-        return self.nvra + ".rpm"
+    repo_key_fields = ("name", "epoch", "version", "release", "arch", "filename")
```

Comment:
So this feels like a half-measure, because really we should make NEVRAs repo-unique and not care about the filename.  This exists pretty much only for the case we ran into like 2 years ago with SUSE storing 2 different source packages for certain libraries for 32-bit and 64-bit optimized builds which had the same NEVRA but different filenames.

But that doesn't really seem like the way it normally goes 90% of the time, normally it's the same NEVRA and the same filename but different locations, so if we start enforcing this we're going to have a lot of repos stop working.

What I'm wondering is if it would be better to make this restriction what we actually want it to be, NEVRA, and then during the sync we try to "just pick one" and sync that - probably the one with the highest build timestamp.

That means we wouldn't sync both of the SUSE source packages but, honestly, at this point I don't care.  It's an edge case of an edge case even for them and there's only so much nonsense we can try to handle at once.

Reply:
But if yes, I need more eyes on https://github.com/pulp/pulp_rpm/pull/2547 since I'd probably want to build on top of that.

Reply:
`What I'm wondering is if it would be better to just use NEVRA as the restriction like we actually want to, and then during the sync we try to "just pick one" throw the others out like DNF would - probably by picking the one with the highest build timestamp or whatever heuristic DNF uses.`

I am down with that however this would not work with 'mirror' option, but as you said - there is a lot of nonsense so difficult to make single usecase and corner case yo work.

Reply:
Yup, but that's what we already do in the mirror case, we just throw up our hands and say sync with a non-mirror mode if they have metadata problems.